mne/preprocessing/stim.py

```diff
@@ -67,3 +67,50 @@ def eliminate_stim_artifact(raw, events, event_id, tmin=-0.005,
         elif mode == 'window':
             raw[picks, first_samp:last_samp] = data * window[np.newaxis, :]
     return raw
+
+def eliminate_stim_artifact_evoked(evoked, mode='linear'):
```

Comment:
you should use `isinstance` to make `eliminate_stim_artifact` accept an `evoked` object instead of copying the code for a new function